--- tcl8.5.6/unix/configure	2008-12-22 14:16:08.000000000 +0000
+++ tcl8.5.6-patched/unix/configure	2009-01-31 16:19:42.495469123 +0000
@@ -8417,7 +8417,7 @@
 
 
 else
-  if test "$arch" = "amd64 i386"; then
+  if test "$arch" = "amd64"; then
 
 		    if test "$GCC" = yes; then
 
@@ -8468,7 +8468,7 @@
 		arch=`isainfo`
 		echo "$as_me:$LINENO: checking whether to use -lsunmath for fp rounding control" >&5
 echo $ECHO_N "checking whether to use -lsunmath for fp rounding control... $ECHO_C" >&6
-		if test "$arch" = "amd64 i386"; then
+		if test "$arch" = "amd64" -o "$arch" = "i386"; then
 
 			echo "$as_me:$LINENO: result: yes" >&5
 echo "${ECHO_T}yes" >&6
@@ -8651,7 +8651,7 @@
 			#CC_SEARCH_FLAGS="${CC_SEARCH_FLAGS},-R,$v9gcclibdir"
 
 else
-  if test "$arch" = "amd64 i386"; then
+  if test "$arch" = "amd64"; then
 
 			SHLIB_LD="$SHLIB_LD -m64 -static-libgcc"
 

